src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -0,0 +1,450 @@
+"""Dataclasses for reflow work."""
+
+
+from itertools import chain
+import logging
+from typing import Iterator, List, Optional, Sequence, Tuple, cast
+from sqlfluff.core.config import FluffConfig
+
+from sqlfluff.core.parser import BaseSegment, RawSegment
+from sqlfluff.core.rules.base import LintFix
+from sqlfluff.utils.reflow.config import ReflowConfig
+from sqlfluff.utils.reflow.depthmap import DepthMap
+
+from sqlfluff.utils.reflow.elements import ReflowElement, ReflowBlock, ReflowPoint
+
+# We're in the utils module, but users will expect reflow
+# logs to appear in the context of rules. Hence it's a subset
+# of the rules logger.
+reflow_logger = logging.getLogger("sqlfluff.rules.reflow")
+
+
+class ReflowSequence:
```

Comment:
Nitpick: Could this be a dataclass? From the constructor, it seems that it could be.

Reply:
originally it was, but running the validation step was awkward. I migrated _away_ from it being dataclass for that reason.